dotCMS/src/integration-test/java/com/dotmarketing/portlets/contentlet/transform/ContentletTransformerTest.java

```diff
@@ -235,8 +269,113 @@ public void Test_Constant_fields_are_in_map() throws Exception {
+                new TestCase(BaseContentType.getBaseContentType(3),
+                        contentletAPI.search("+basetype:3", limit, 0, null, user, respectFrontEndUsers)),
+                new TestCase(BaseContentType.getBaseContentType(4),
+                        contentletAPI.search("+basetype:4", limit, 0, null, user, respectFrontEndUsers)),
+                new TestCase(BaseContentType.getBaseContentType(5),
+                        contentletAPI.search("+basetype:5", limit, 0, null, user, respectFrontEndUsers)),
+                new TestCase(BaseContentType.getBaseContentType(6),
+                        contentletAPI.search("+basetype:6", limit, 0, null, user, respectFrontEndUsers)),
+                new TestCase(BaseContentType.getBaseContentType(7),
+                        contentletAPI.search("+basetype:7", limit, 0, null, user, respectFrontEndUsers)),
+                new TestCase(BaseContentType.getBaseContentType(8),
+                        contentletAPI.search("+basetype:8", limit, 0, null, user, respectFrontEndUsers))
+        };
+    }
+
+    private static class TestCase {
+        final BaseContentType baseContentType;
+        final List<Contentlet> contentlets;
+
+        TestCase(final BaseContentType baseContentType,
+                final List<Contentlet> contentlets) {
+            this.baseContentType = baseContentType;
+            this.contentlets = contentlets;
+        }
+    }
+
+    @Test
+    @UseDataProvider("listTestCases")
+    public void Transformer_Backwards_Compatibility_Test(final TestCase testCase) {
+
+        final List <Contentlet> list = testCase.contentlets;
+        final String baseTypeName = testCase.baseContentType.name();
+
+        if(list.isEmpty()){
+           Logger.warn(ContentletTransformerTest.class, String.format("Unable to get test samples of type `%s`",testCase.baseContentType.name()));
+           return;
+        }
+
+        final ContentletToMapTransformer contentletToMapTransformer = new ContentletToMapTransformer(list);
+
+        final List<Map<String, Object>> transformedList1 = contentletToMapTransformer.toMaps();
+
+        final DotTransformer dotTransformer = new Builder().defaultOptions().content(list).build();
+
+        final List<Map<String, Object>> transformedList2 = dotTransformer.toMaps();
+
+        assertEquals(transformedList1.size(), transformedList2.size());
+        for (int i = 0; i < list.size(); i++) {
+
+            final Contentlet original = list.get(i);
+            //We compare the transformation results are the same using the default options on the new Transformer
+            final Map<String, Object> objectMap1 = transformedList1.get(i);
+
+            final Map<String, Object> objectMap2 = transformedList2.get(i);
+
+            assertTrue(String.format(" baseType `%s` should have same (or more) number of properties " ,baseTypeName),objectMap2.size() >= objectMap1.size());
+
+            for (String key1 : objectMap1.keySet()) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'key1' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5596402)